plugin.video.vtm.go/resources/lib/kodilogging.py

```diff
@@ -0,0 +1,50 @@
+# -*- coding: utf-8 -*-
+"""Log handler for Kodi"""
+
+from __future__ import absolute_import, division, unicode_literals
+
+import logging
+
+import xbmc
+import xbmcaddon
+
+ADDON = xbmcaddon.Addon()
+
+
+class KodiLogHandler(logging.StreamHandler):
+    """ A log handler for Kodi """
+
+    def __init__(self):
+        logging.StreamHandler.__init__(self)
+        formatter = logging.Formatter("[{}] [%(name)s] %(message)s".format(ADDON.getAddonInfo("id")))
+        self.setFormatter(formatter)
+
+    def emit(self, record):
+        """ Emit a log message """
+        levels = {
+            logging.CRITICAL: xbmc.LOGFATAL,
+            logging.ERROR: xbmc.LOGERROR,
+            logging.WARNING: xbmc.LOGWARNING,
+            logging.INFO: xbmc.LOGNOTICE,
+            logging.DEBUG: xbmc.LOGDEBUG,
+            logging.NOTSET: xbmc.LOGNONE,
+        }
+
+        # Map DEBUG level to LOGNOTICE if debug logging setting has been activated
+        # This is for troubleshooting only
+        if ADDON.getSetting('debug_logging') == 'true':
```

Comment:
I think this one is rather expensive to call for each log line? Would it not be better to call it in the constructor and set it the log level there?

Reply:
I've benchmarked this on my laptop:

```py
        start = time.time()
        for x in range(1000):
            ADDON.getSetting('debug_logging')
        end = time.time()
        _LOGGER.error('time = %s', end-start)

```

This returns:
```
2020-07-22 23:02:03.015 T:140123968890624   ERROR: [plugin.video.vtm.go] [menu] time = 0.00075888633728

``` 

This seems perfectly fine. If I do this in the constructor, I need to have code to invalidate this when the settings change.

Reply:
I had performance issues with my Raspberry Pi's due to this kind of code. So it's not wrong, just a point of attention.